hrjobcontract/CRM/Hrjobcontract/Upgrader.php

```diff
@@ -1348,6 +1336,84 @@ public function upgrade_1040() {
     return TRUE;
   }
   
+  /**
+   * Renames option group title from job contract end reason
+   * to contract end reasons
+   *
+   * @return bool
+   */
+  public function upgrade_1041() {
+    civicrm_api3('OptionGroup', 'get', [
+      'name' => 'hrjc_contract_end_reason',
+      'api.OptionGroup.create' => [
+        'id' => '$value.id',
+        'title' => 'Contract End Reasons'
+      ],
+    ]);
+    
+    return TRUE;
+  }
+  
+  /**
+   * Deletes cost centre other option value or disable it if not in use
+   *
+   * @return bool
+   */
+  public function upgrade_1042() {
+    $jobRoles = civicrm_api3('HrJobRoles', 'get');
+    if ($jobRoles['count'] == 0) {
+      $this->deleteCostCentreOther();
+    }
+  
+    $otherId = $this->retrieveCostCentreOtherId();
+    $inUse = FALSE;
+    $roles = $jobRoles['values'];
+    $pattern = '/(\|' . $otherId . '\|)/';
+    foreach ($roles as $role) {
+      if (preg_match($pattern, $role['cost_center'])) {
+        $inUse = TRUE;
+        break;
+      }
+    }
+    
+    if (! $inUse) { // disables cost centre other
+      civicrm_api3('OptionValue', 'get', [
+        'option_group_id' => 'cost_centres',
+        'name' => 'Other',
+        'api.OptionValue.create' => ['id' => '$value.id', 'is_active' => 0],
+      ]);
+    }
+    
+    return TRUE;
+  }
+  
+  /**
+   * Deletes cost centre option value with name other
+   */
+  private function deleteCostCentreOther() {
+    civicrm_api3('OptionValue', 'get', [
+      'option_group_id' => 'cost_centres',
+      'api.OptionValue.delete' => [
+        'id' => '$value.id',
+        'name' => 'Other'
+      ],
+    ]);
+  }
+  
+  /**
+   * Fetches the id of cost center other option value
```

Comment:
It's just a small point, but in places where you mention "other" option value, can you put "other" in quotes? It's a bit confusing otherwise

Reply:
Noted. thanks.